odoo/tools/image.py

```diff
@@ -113,6 +114,10 @@ def image_quality(self, quality=0, output_format=''):
         :return: image
         :rtype: bytes or False
         """
+        if odoo.http.request and (quality == -1 or \
+            odoo.http.request.env['ir.config_parameter'].sudo().get_param('global.no_quality_optimization', '')):
```

Comment:
Please don't do that, the tools are meant to be utility functions that are stateless and called by the business logic. They should do what the caller asks, not decide themselves if they're going to do something or not - that should be part of the business logic.

That circular dependency was a good hint. Deal with this at the callsites instead.